migration/idempotent/999-last.sql

```diff
@@ -0,0 +1,271 @@
+                else ''
+            end)
+        FROM pg_catalog.pg_depend d
+        INNER JOIN pg_catalog.pg_extension e ON (d.refobjid = e.oid)
+        INNER JOIN pg_catalog.pg_class k ON (d.objid = k.oid)
+        INNER JOIN pg_namespace n ON (k.relnamespace = n.oid)
+        WHERE d.refclassid = 'pg_catalog.pg_extension'::pg_catalog.regclass
+        AND d.deptype = 'e'
+        AND e.extname = 'promscale'
+        AND k.relkind IN ('r', 'p', 'S') -- tables AND sequences
+        AND (n.nspname, k.relname) NOT IN
+        (
+            -- these should NOT be config tables
+            -- migration table will be populated by the installation of extension
+            ('_ps_catalog'::name, 'migration'::name),
+            -- we want the telemetry data reset with restores
+            ('_ps_catalog'::name, 'promscale_instance_information'::name)
+        )
+        ORDER BY n.nspname, k.relname
+    )
+    LOOP
+        EXECUTE _sql;
+    END LOOP;
+END;
+$block$;
+
+-- make sure prom_admin can do a dump/restore
+DO $$
+DECLARE
+    _rec record;
+    _schema text;
+    _schemas text[] = ARRAY
+    [ '_prom_catalog'
+    , '_prom_ext'
+    , '_ps_catalog'
+    , '_ps_trace'
+    , 'prom_api'
+    , 'prom_data'
+    , 'prom_data_exemplar'
+    , 'prom_data_series'
+    , 'prom_info'
+    , 'prom_metric'
+    , 'prom_series'
+    , 'ps_tag'
+    , 'ps_trace'
+    ];
+BEGIN
+    -- grant schema privileges
+    FOREACH _schema IN ARRAY _schemas
+    LOOP
+        EXECUTE format('ALTER DEFAULT PRIVILEGES IN SCHEMA %I GRANT ALL PRIVILEGES ON TABLES TO prom_admin', _schema);
+        EXECUTE format('ALTER DEFAULT PRIVILEGES IN SCHEMA %I GRANT ALL PRIVILEGES ON SEQUENCES TO prom_admin', _schema);
+        EXECUTE format('GRANT USAGE, CREATE ON SCHEMA %I TO prom_admin', _schema);
+    END LOOP;
+
+    -- grant table/sequence privileges
+    FOR _rec IN
+    (
+        SELECT
+            n.nspname,
+            k.relname,
+            case k.relkind
+                when 'S' then 'SEQUENCE'
+                else 'TABLE'
+            end as kind
+        FROM pg_catalog.pg_class k
+        INNER JOIN pg_namespace n ON (k.relnamespace = n.oid)
+        WHERE k.relkind IN ('r', 'p', 'S') -- tables AND sequences
+        AND n.nspname = ANY(_schemas)
+        ORDER BY k.relkind, n.nspname, k.relname
+    )
+    LOOP
+        EXECUTE format('GRANT ALL PRIVILEGES ON %s %I.%I TO prom_admin', _rec.kind, _rec.nspname, _rec.relname);
+    END LOOP;
+
+    ALTER TABLE _ps_trace.span OWNER TO prom_admin; --required for restores to attach child tables
+    ALTER TABLE _ps_trace.link OWNER TO prom_admin; --required for restores to attach child tables
+    ALTER TABLE _ps_trace.event OWNER TO prom_admin; --required for restores to attach child tables
+    ALTER TABLE _prom_catalog.series OWNER TO prom_admin; --required for restores to attach child tables
+
+END$$;
+
+-- make sure prom_admin can do a dump/restore
+DO $$
```

Comment:
hmm why do we need to do this, won't tsdbadmin have this already granted by the after-create scripts?

Reply:
yes. on cloud. but not for folks using the docker images or the linux packages, right?

Reply:
Yeah but that seems like a timescaledb issue. In the ha-docker image it should work because after-create script is part of that. And on-prem people usually have superuser.

Fundamentally, I am a bit uncomfortable setting permissions on internal schemas of another extension even if it is timescaledb

Reply:
I think for non-superuser dump restore we have to just target cloud

Reply:
> Fundamentally, I am a bit uncomfortable setting permissions on internal schemas of another extension even if it is timescaledb

yeah, agreed.